### ISSUE TITLE
Re-consider branch and release strategy of scalameta/metals

### ISSUE THREAD

**Issue by tanishiking** [scalameta/metals-feature-requests#300]
## TL;DR;
- Users have to wait until the next release of Metals before using the latest version of Scala(3)
- Maybe we want to adapt Metals to the new release (or branch) strategy so that Metals can offer the latest Scala support to end-users (even people who don't use the pre-release versions).

## Problem

Currently, `scalameta/metals` follows a branch strategy called `GitHub-flow`.

In this strategy, all the features/bugfixes go into the `main` branch, and we expect the `main` branch is ALWAYS deployable. 
On top of that, in this branching strategy, hotfix also goes into the `main` branch as other features or normal bug fixes.

> Normally, you would have to do a 'hotfix' or something outside of the normal process, but it's simply part of our normal process - there is no difference in the GitHub flow between a hotfix and a very small feature.
https://githubflow.github.io/

This is fine if the project release a new version immediately after the hotfix is merged into the `main` branch.

However, in Metals,

- We usually release Metals once a month
- the `main` branch is not necessarily "deployable"
  - the `main` branch sometimes contains a bit unstable feature, and Metals developers (and some developers from the community) test the pre-release version before release.

As a result, Metals have to wait for the next release before shipping the hotfixes. The problem is that Metals usually want to "hotfix," adding support for the new Scala compilers.
For example, Scala 3.2.0 is released these days, and Metals already support it in the pre-release versions. However, normal Metals users (who don't use the pre-release version) must wait until the next Metals release.

**I believe Metals should release a new version that supports the latest Scala version as soon as possible.**

### Describe the solution you'd like

Maybe we want to adopt something like `github-flow + release branch (for hotfixes)` like [sbt does](https://github.com/sbt/sbt/blob/1f29c9053cdc52128fa45cfe0b22e9a4d085cc1b/DEVELOPING.md#branch-to-work-against)

We will have two branches

- `main`
  - When we work on adding new features or normal bug fixes (including hotfixes, supporting new Scala version), fork from / merge into this branch.
    - Same as we do now.
   - we should release new version of Metals for the "normal monthly release".
- stable branch `0.$MINOR.x` branch
  - when we release a new version of Metals from the `main` branch, fork from the `main`
     - (for example, if we release `0.12.0`, fork from the `main` branch to `0.12.x`.
  - hotfixes to the `main` branch should be backported to the latest stable branch.
  - We should release all changes in the stable branch immediately
    - incrementing patch version
    - without writing release notes
  - Once the new minor version of Metals is released, we can delete the existing stable branch and fork another stable branch from `main`.

As you might notice, in this release strategy, we will increment **minor versions** for the normal monthly release, and increment patch versions only for hotfixes.

### Describe alternatives you've considered

- Release from the `main` branch once we merge the hotfixes or support new Scala versions.
  - The problem is, that the latest `main` is not necessarily ready for review.
  - We won't have enough time for writing a release note.
    - Alternatively, release the features silently, and write a release note at some point.
- Keep current release cycles

### Additional context

This proposal came up to my mind because of the conversation on Discord by @Swoorup and @SethTisue

```
sytherax — 09/02/2022
Anyone tried 3.2.0? 

getting 
2022.09.02 02:53:47 ERROR Failed to fetch mtags for 3.2.0
Error downloading org.scalameta:mtags_3.2.0:0.11.8+54-8ab4afa8-SNAPSHOT
  not found: https://repo1.maven.org/maven2/org/scalameta/mtags_3.2.0/0.11.8+54-8ab4afa8-SNAPSHOT/mtags_3.2.0-0.11.8+54-8ab4afa8-SNAPSHOT.pom
  not found: https://oss.sonatype.org/content/repositories/public/org/scalameta/mtags_3.2.0/0.11.8+54-8ab4afa8-SNAPSHOT/mtags_3.2.0-0.11.8+54-8ab4afa8-SNAPSHOT.pom
  not found: https://oss.sonatype.org/content/repositories/snapshots/org/scalameta/mtags_3.2.0/0.11.8+54-8ab4afa8-SNAPSHOT/mtags_3.2.0-0.11.8+54-8ab4afa8-SNAPSHOT.pom
nvm new build fixed it

SethTisue — 09/02/2022
in general, tooling may not be available until the release is announced (as opposed to merely published)
```

https://discord.com/channels/632642981228314653/632652693013528589/1014958307829620817

### Search terms

release, branch strategy, latest scala version

**Comment by tgodzik** [comment]
Actually, it's just that snapshot that doesn't have the new support added. We already added support for 3.2.0 in `0.11.8` and the latest snapshots. 

> Users have to wait until the next release of Metals before using the latest version of Scala(3)

They don't actually have to wait currently, we release for past versions and current snapshots

**Comment by tanishiking** [comment]
Yeah, I was misunderstanding, and had a same conversation in Discord https://discord.com/channels/632642981228314653/718047292749381662/1017009737683963924 😆

**Comment by tanishiking** [comment]
While hotfix stuff is still a kinda problem, but actually my main point was supporting the latest Scala compilers 😅 So now I'm not sure about whether we should change the branch strategy.
So closing for now
